rust/lance-encoding/src/data.rs

```diff
@@ -893,6 +953,17 @@ impl DataBlock {
     as_type_ref!(as_variable_width_ref, VariableWidth, VariableWidthBlock);
     as_type_ref!(as_struct_ref, Struct, StructDataBlock);
     as_type_ref!(as_dictionary_ref, Dictionary, DictionaryDataBlock);
+    as_type_mut_ref!(as_all_null_mut_ref, AllNull, AllNullDataBlock);
+    as_type_mut_ref!(as_nullable_mut_ref, Nullable, NullableDataBlock);
+    as_type_mut_ref!(as_fixed_width_mut_ref, FixedWidth, FixedWidthDataBlock);
+    as_type_mut_ref!(
+        as_fixed_size_list_mut_ref,
+        FixedSizeList,
+        FixedSizeListBlock
+    );
+    as_type_mut_ref!(as_variable_width_mut_ref, VariableWidth, VariableWidthBlock);
+    as_type_mut_ref!(as_struct_mut_ref, Struct, StructDataBlock);
+    as_type_mut_ref!(as_dictionary_mut_ref, Dictionary, DictionaryDataBlock);
```

Comment:
Can we call the function / macro name `..._mut` and not `..._mut_ref` for consistency with things like `[Option](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_mut)`